hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -1061,8 +1061,14 @@ public static SingularAssociationAttributeMapping buildSingularAssociationAttrib
 						sessionFactory
 				);
 
-		final FetchTiming fetchTiming = FetchStrategyHelper.determineFetchTiming( fetchStyle, type, sessionFactory );
+		final FetchTiming fetchTiming;
 
+		if ( !( (ToOne) bootProperty.getValue() ).isLazy() ) {
+			fetchTiming = FetchTiming.IMMEDIATE;
+		}
+		else {
+			fetchTiming = FetchStrategyHelper.determineFetchTiming( fetchStyle, type, sessionFactory );
+		}
```

Comment:
Feel free to ignore my comment. It would be great to centralize all the fetch timing decision logic into `FetchStrategyHelper` for code maintaining purpose.

Reply:
I don't think that is really possible.  Happy to be proven wrong because, as you say, it would make it better in all the ways that centralizing code does.

Reply:
To be clear, anything is *possible*, sure.  The question is whether centralizing this make the code more simple or more complex. Specifically does centralizing it mean passing more arguments?  How many more?  There is a tipping point.